# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/svg/SVGRect.h
# Commit: 61a21b2c9e6e
# Full Hash: 61a21b2c9e6e9df619cf74f656d3989250939414
# Author: longsonr <longsonr@gmail.com>
# Date: 2019-08-20 09:38:33
# Description:
#   Bug 1572904 - refactor cycle collection code after DOMSVGAngle r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D42453
# ==============================================================================

diff -r b85cb0cd1e49 -r 61a21b2c9e6e dom/svg/SVGRect.h
--- a/dom/svg/SVGRect.h	Mon Aug 19 23:21:27 2019 +0000
+++ b/dom/svg/SVGRect.h	Mon Aug 19 19:10:03 2019 +0000
@@ -18,12 +18,12 @@
 
 class SVGSVGElement;
 
-class SVGRect final : public nsISupports, public nsWrapperCache {
+class SVGRect final : public nsWrapperCache {
  public:
   typedef enum { BaseValue, AnimValue, CreatedValue } RectType;
 
-  NS_DECL_CYCLE_COLLECTING_ISUPPORTS
-  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS(SVGRect)
+  NS_INLINE_DECL_CYCLE_COLLECTING_NATIVE_REFCOUNTING(SVGRect)
+  NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_NATIVE_CLASS(SVGRect)
 
   /**
    * Generic ctor for objects that are created for an attribute.